frame/system/src/lib.rs

```diff
@@ -524,6 +524,7 @@ pub mod pallet {
 
 	/// Event for the System pallet.
 	#[pallet::event]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```

Comment:
I don't really mind the `Serialize` and `Deserialize` derives on the other types, but if it's on the `Event` type, then I think the `#[pallet::event]` macro should be generating the derives during macro expansion, since it's not just the FRAME system/support pallets that contain the `Event` type. The code for the pallet event macro expansion is located in `frame/support/procedural/pallet/expand/event.rs`.